communication/vehicleconnections/mavsdkvehicleconnection.cpp

```diff
@@ -32,10 +37,75 @@ MavsdkVehicleConnection::MavsdkVehicleConnection(std::shared_ptr<mavsdk::System>
         mVehicleState->setName("Copter " + QString::number(mSystem->get_system_id()));
         break;
     case MAV_TYPE_GROUND_ROVER:
-        qDebug() << "MavsdkVehicleConnection: we are talking to a MAV_TYPE_GROUND_ROVER / WayWise.";
-        // TODO: detect different rover types...
-        mVehicleState = QSharedPointer<CarState>::create(mSystem->get_system_id());
-        mVehicleState->setName("Car " + QString::number(mSystem->get_system_id()));
+        if (mWaywiseObjectType == WAYWISE_OBJECT_TYPE_TRUCK) {
```

Comment:
all of these if statement will be simplify using a switch and the code will be more readable , also the compiler optimizes switch cases using a jump table which is constant-time complexity , where is statements can lead to long execution times due to miss-branch predictions

Reply:
Here we already know the mav type as ground rover and only need to check whether the object type is truck or a car. Using a switch here could be misleading.